packages/agreement-process/src/services/agreementService.ts

```diff
@@ -437,12 +438,14 @@ async function createAndCopyDocumentsForClonedAgreement(
   const docs = await Promise.all(
     clonedAgreement.consumerDocuments.map(async (d) => {
       const newId: AgreementDocumentId = generateId();
+      const documentDestinationPath = `${config.consumerDocumentsPath}`;
```

Comment:
Here according to the Scala codebase, you should also add the agreement ID in the base path of the destination, doing something like `${config.consumerDocumentsPath}/${clonedAgreement.id}` https://github.com/pagopa/interop-be-agreement-process/blob/0cdda61aa22b466af59a38528c452be0a8d3533d/src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala#L1092-L1094

I know it's different from what happens in the catalog, but also the two Scala implementation are different apparently. In the catalog, documents are stored in paths like `catalogDocumentsPath/documentId/doc.pdf` while in the agreement Scala seems to store them in paths like `agreementDocumentsPath/agreementId/documentId/doc.pdf`.

Maybe @galales knows more about why there is this difference?

Reply:
I will also fix the delete tests accordingly - even though they just contain mocked paths, it's good to have them resemble the real ones

Reply:
as you suggest [here](https://github.com/pagopa/interop-be-agreement-process/blob/0cdda61aa22b466af59a38528c452be0a8d3533d/src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala#L1092-L1094) the destination file is different, my fault I had different path composition fixed in mind.